src/tests/HDFFile.cxx

```diff
@@ -66,6 +66,116 @@ class T_CommandHandler : public testing::Test {
     return event_index[i2] == cue_index;
   }
 
+  static void create_static_file_with_hdf_output_prefix() {
+    using namespace FileWriter;
+    using std::array;
+    using std::vector;
+    using std::string;
+    MainOpt &main_opt = *g_main_opt.load();
+    std::string hdf_output_prefix = "tmp-relative-output";
+    {
+#ifdef _MSC_VER
+      // NOTE just a suggestion, not tested on windows so far, we have no
+      // windows here.
+      _mkdir(hdf_output_prefix.c_str());
+#else
+      mkdir(hdf_output_prefix.c_str(), 0777);
+#endif
+    }
+    {
+      std::string jsontxt =
+          fmt::format(R""({{"hdf-output-prefix": "{}"}})"", hdf_output_prefix);
+      rapidjson::Document cfg;
+      cfg.Parse(jsontxt.c_str());
+      main_opt.config_file = merge(cfg, main_opt.config_file);
+      main_opt.hdf_output_prefix = hdf_output_prefix;
+    }
+    rapidjson::Document json_command;
+    {
+      using namespace rapidjson;
+      auto &j = json_command;
+      auto &a = j.GetAllocator();
+      j.SetObject();
+      Value nexus_structure;
+      nexus_structure.SetObject();
+
+      Value children;
+      children.SetArray();
+
+      {
```

Comment:
There are quite a few lines of json which need to be there but which aren't actually relevant to the test and make it less obvious to the reader what is being tested. Would it be possible to have a "core" json description in a function to which you add the specific bit you want to test in each test case?